### DISCUSSION TITLE
How to use -elset in M module?

### DISCUSSION THREAD

**Discussion by smallrain1996** [neperfepx/neper#412] [Q&A] [answered]
How do I use the -elset function to create a collection of elements that meet specific requirements? 
I tried the following code, but without success.
```
neper -T -n 100 -dim 2 -o test
neper -M test.tess -elset "y_top:y>0.5" -format inp -o test
```
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.4.2-5
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/smallrain/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] test.tess -elset y_top:y>0.5 -format inp -o test
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `test.tess'...
Info   :     [i] Parsed file `test.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.025) 
Info   :   - 0d meshing... 100%
Info   :   - 1d meshing... 100%
Info   :   - 2d meshing... 100% (0.72|0.91/98%| 0%| 2%) 
Info   : Searching nsets and fasets...
Info   : Searching elsets...
Info   :   - y_top...
Info   : Writing mesh results...
Info   :   - Preparing mesh...
Info   :   - Mesh properties:
Info   :     > Node number:     2518
Info   :     > Elt  number:     4850
Info   :     > Mesh area:    1.000
Info   :   - Writing mesh...
Info   :     [o] Writing file `test.inp'...
Info   :     [o] Wrote file `test.inp'.
free(): double free detected in tcache 2
Aborted (core dumped)
```
Could you please provide a use case for the -elset function, thanks

**Comment by rquey** [comment, marked as answer]
This seems largely buggy, indeed. Why do you need this capability for?

I can't remember why I added it, and there is now `-entity` in `-S`, which is similar (but only for post-processing).

**Reply by smallrain1996** [reply]
Thank you for your reply, I needed this functionality because I wanted to set up a set of elements in the INP file that met certain conditions to facilitate the subsequent application of loads to this set or to count the simulation results. Now I can also implement my need to construct my own elset based on the information of the individual elements.

**Comment by ashokambig** [comment]
Dear Quey,

Do we have solution to create elsets of our wish?
I am currently working on this. This is an important feature for my simulation. Here I am trying to identify small grains based on their equivalent diameter. This set will have a specific material property in my simulation. I tried to run the attached script. But there is an error. I see that elsets are created, but somehow I dont seem to understand this command properly. Can you please help? I have attached the errod message and the script for your reference.

Best regards
Ashok


========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.2.0
Info   : Built with: gsl|muparser|opengjk|nlopt|libscotch
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2021, and GNU GPL'd, by Romain Quey.
Info   : Ignoring initialization file.
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] abq_input.tess -elttype hex -order 2 -elset
         A_grain:diameq<0.088 -faset none -format inp
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `abq_input.tess'...
Info   :     [i] Parsed file `abq_input.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.02154) 100%

Info   :   - 3D meshing... 100%
Info   :   - Switching mesh to order 2...
Info   : Searching nsets and fasets...
Info   : Searching elsets...
Info   :   - A_grain...
Info   : Writing mesh results...
Info   :   - Preparing mesh...
Info   :   - Mesh properties:
Info   :     > Node number:   408665
Info   :     > Elt  number:    97336
Info   :     > Mesh volume:    1.000
Info   :   - Writing mesh...
Info   :     [o] Writing file `abq_input.inp'...
Info   :     [o] Wrote file `abq_input.inp'.
./test: line 4:  5985 Segmentation fault      (core dumped) neper --rcfile none -M abq_input.tess -elttype hex -order 2 -elset "A_grain:diameq<0.088" -faset none -format inp
[test.txt](https://github.com/neperfepx/neper/files/10945330/test.txt)
